sklearn/tree/_criterion.pxd

```diff
@@ -10,55 +10,52 @@
-from ._tree cimport INT32_t          # Signed 32 bit integer
-from ._tree cimport UINT32_t         # Unsigned 32 bit integer
+from ..utils._typedefs cimport float64_t, intp_t
+
 
 cdef class Criterion:
     # The criterion computes the impurity of a node and the reduction of
     # impurity of a split on that node. It also computes the output statistics
     # such as the mean in regression and class probabilities in classification.
 
     # Internal structures
-    cdef const DOUBLE_t[:, ::1] y         # Values of y
-    cdef const DOUBLE_t[:] sample_weight  # Sample weights
+    cdef const float64_t[:, ::1] y         # Values of y
+    cdef const float64_t[:] sample_weight  # Sample weights
 
-    cdef const SIZE_t[:] sample_indices   # Sample indices in X, y
-    cdef SIZE_t start                     # samples[start:pos] are the samples in the left node
-    cdef SIZE_t pos                       # samples[pos:end] are the samples in the right node
-    cdef SIZE_t end
-    cdef SIZE_t n_missing                # Number of missing values for the feature being evaluated
-    cdef bint missing_go_to_left         # Whether missing values go to the left node
+    cdef const intp_t[:] sample_indices    # Sample indices in X, y
+    cdef intp_t start                      # samples[start:pos] are the samples in the left node
+    cdef intp_t pos                        # samples[pos:end] are the samples in the right node
+    cdef intp_t end
+    cdef intp_t n_missing                  # Number of missing values for the feature being evaluated
+    cdef bint missing_go_to_left           # Whether missing values go to the left node
 
-    cdef SIZE_t n_outputs                 # Number of outputs
-    cdef SIZE_t n_samples                 # Number of samples
-    cdef SIZE_t n_node_samples            # Number of samples in the node (end-start)
-    cdef double weighted_n_samples        # Weighted number of samples (in total)
-    cdef double weighted_n_node_samples   # Weighted number of samples in the node
-    cdef double weighted_n_left           # Weighted number of samples in the left node
-    cdef double weighted_n_right          # Weighted number of samples in the right node
-    cdef double weighted_n_missing       # Weighted number of samples that are missing
+    cdef intp_t n_outputs                  # Number of outputs
+    cdef intp_t n_samples                  # Number of samples
+    cdef intp_t n_node_samples             # Number of samples in the node (end-start)
+    cdef double weighted_n_samples         # Weighted number of samples (in total)
+    cdef double weighted_n_node_samples    # Weighted number of samples in the node
+    cdef double weighted_n_left            # Weighted number of samples in the left node
+    cdef double weighted_n_right           # Weighted number of samples in the right node
+    cdef double weighted_n_missing         # Weighted number of samples that are missing
```

Comment:
Could `double` (here and also elsewhere) be changed to `float64_t`?

Reply:
I second this suggestion.

Reply:
A separate PR would be easier to review.

Reply:
In that case, I'll leave this as is for now. Would this PR be mergable? I can help test and make the systematic change in the tree submodule once we resolve this PR?